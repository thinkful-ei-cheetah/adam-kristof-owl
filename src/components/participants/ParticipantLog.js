import React from 'react';
import './ParticipantLog.css';


export default function ParticipationLog (props){
    console.log(props);
    const participants = props.participants.map(participant => {
        return (
          <li key={participant.id}>
            <div className='personContainer'>
              <img src={participant.avatar} alt={`avitar for ${participant.name}`} />
              <div className='personName'>
                <div>{participant.name}</div>
                <div>{participant.onStage ? 'on stage' : 'in session'}</div>
              </div>
            </div>
          </li>
        )
      })

    return (
        <div className="participentView">
            <ul>
                {participants}
            </ul>
        </div>
    )

}
