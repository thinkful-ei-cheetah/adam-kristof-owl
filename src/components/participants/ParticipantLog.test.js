import React from 'react';
import ReactDOM from 'react-dom';
import ParticipantLog from './ParticipantLog'
import renderer from 'react-test-renderer'  


describe('Participant Componenet working', ()=>{
  it('Renders to the page without crashing', ()=>{
    const div = document.createElement('div')
    ReactDOM.render(<ParticipantLog  participants={[]}/>, div)
    ReactDOM.unmountComponentAtNode(div)
  });
  
  it('Renders as it did the first time!', ()=>{
    const tree = renderer
    .create(<ParticipantLog participants={[]}/>)
    .toJSON()
    expect(tree).toMatchSnapshot()  
  ;
  });
});
