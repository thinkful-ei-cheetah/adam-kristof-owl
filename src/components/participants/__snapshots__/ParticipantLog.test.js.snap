// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Participant Componenet working Renders as it did the first time! 1`] = `
<div
  className="participentView"
>
  <ul />
</div>
`;
